src/vcpkg/base/downloads.cpp

```diff
@@ -672,107 +132,168 @@ namespace vcpkg
         return true;
     }
 
-    static std::vector<int> curl_bulk_operation(DiagnosticContext& context,
-                                                View<Command> operation_args,
-                                                StringLiteral prefixArgs,
-                                                View<std::string> headers,
-                                                View<std::string> secrets)
+    struct CurlRequestPrivateData
+    {
+        size_t request_index = 0;
+        std::unique_ptr<WriteFilePointer> file = nullptr;
+    };
+    static size_t write_file_callback(void* contents, size_t size, size_t nmemb, void* param)
+    {
+        auto* file = reinterpret_cast<WriteFilePointer*>(param);
+        if (!file) return 0;
+        return file->write(contents, size, nmemb);
```

Comment:
```suggestion
        if (!param) return 0;
        return static_cast<WriteFilePointer*>(param)->write(contents, size, nmemb);
```

to/from `void*` only needs `static_cast`, not `reintrepret_cast`. Also the null check can be done first to avoid needing an extra variable.